src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs

```diff
@@ -22,7 +22,14 @@ internal class ProjectRestoreRequest
             ExistingLockFile = existingLockFile;
             MaxDegreeOfConcurrency = request.MaxDegreeOfConcurrency;
             Project = packageSpec;
-            PackageExtractionContext = request.PackageExtractionContext;
+            PackageExtractionContext = new PackageExtractionContext(
+                request.PackageSaveMode,
+                request.XmlDocFileSaveMode,
+                request.ClientPolicyContext,
+                log)
```

Comment:
this is so confusing to understand now like where which logger is being used? Could this logger thing be simplified?

Reply:
I don't know enough about this code to propose a way to simplify it. As far as I understand every code path has only one logger. If it doesn't have one, but has access to a project it creates a `LoggerAdapter` to pass as an `ILogger`. The only case that will have more than one is when the `ICollectorLogger` is around, which should be preferred over any other logger.

The cause of this issue was that by creating `PackageExtractionContext` at the beginning and passing it down, we were creating it before the `RestoreCollectorLogger` was created, therefore `PackageExtractionContext` ended up with the wrong logger.